registry/registry.go

```diff
@@ -123,6 +125,23 @@ func (r *RouteRegistry) register(uri route.Uri, endpoint *route.Endpoint) route.
 	}
 
 	endpointAdded := pool.Put(endpoint)
+	//Check endpoint for a load balancing algorithm. If it does exist & differs from that of the pool, then take the lb algorithm of the endpoint, otherwise keep the pools lb algorithm.
```

Comment:
Comments should be folded at a reasonable length and always start with a space.

Reply:
done